src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -47,8 +47,7 @@ public class BlockTree : IBlockTree
         private readonly LruCache<Keccak, BlockHeader> _headerCache = new LruCache<Keccak, BlockHeader>(CacheSize);
         private readonly LruCache<long, ChainLevelInfo> _blockInfoCache = new LruCache<long, ChainLevelInfo>(CacheSize);
 
-        private const int MaxQueueSize = 10_000_000;
-
+        private const int BestKnownSearchLimit = 256_000_000;
```

Comment:
Maybe make it configurable? 
Or store the actual level index in DB, making this binary search obsolete?